pkg/lib/credentials.js

```diff
@@ -277,56 +277,56 @@ function Keys() {
         let failure = _("Not a valid private key");
         let sent_password = false;
 
-        return new Promise((resolve, reject) => {
-            const proc = cockpit.spawn(["ssh-add", name],
-                                       { pty: true, environ: ["LC_ALL=C"], err: "out", directory: self.path });
-
-            const timeout = window.setTimeout(() => {
-                failure = _("Prompting via ssh-add timed out");
-                proc.close("terminated");
-            }, 10 * 1000);
-
-            proc
-                    .stream(data => {
-                        buffer += data;
-                        output += data;
-                        if (perm_exp.test(buffer)) {
-                            failure = _("Invalid file permissions");
-                            buffer = "";
-                        } else if (ask_exp.test(buffer)) {
-                            buffer = "";
-                            failure = _("Password not accepted");
-                            proc.input(password + "\n", true);
-                            sent_password = true;
-                        } else if (bad_exp.test(buffer)) {
-                            buffer = "";
-                            proc.input("\n", true);
-                        }
-                    })
-                    .then(() => {
-                        refresh();
-                        resolve();
-                    })
-                    .catch(ex => {
-                        console.log(output);
-                        if (ex.exit_status)
-                            ex = new Error(failure);
-
-                        ex.sent_password = sent_password;
-                        reject(ex);
-                    })
-                    .finally(() => window.clearInterval(timeout));
+        await self.p_have_path;
+
+        const proc = cockpit.spawn(["ssh-add", name],
+                                   { pty: true, environ: ["LC_ALL=C"], err: "out", directory: self.path });
+
+        const timeout = window.setTimeout(() => {
```

Comment:
Would have been nice to also see this timeout block moved.  Sorry for failing to mention it on the last review.

Reply:
Hmm. I don't have a strong opinion, but I actually like defining timeouts very close to the open, so that they are less prone to be short-circuited by some conditional.